pdf/contentstream/inline-image.go

```diff
@@ -437,14 +438,41 @@ func (csp *ContentStreamParser) ParseInlineImage() (*ContentStreamInlineImage, e
 					} else if state == 3 {
 						skipBytes = append(skipBytes, c)
 						if core.IsWhiteSpace(c) {
-							// image data finished.
-							if len(im.stream) > 100 {
-								common.Log.Trace("Image stream (%d): % x ...", len(im.stream), im.stream[:100])
-							} else {
-								common.Log.Trace("Image stream (%d): % x", len(im.stream), im.stream)
+							// Whitspace after EI.
+							// To ensure that is not a part of encoded image data: Peek up to 20 bytes ahead
+							// and check that the following data is valid objects/operands.
+							peekbytes, err := csp.reader.Peek(20)
+							if err != nil && err != io.EOF {
+								return nil, err
+							}
+							dummyParser := NewContentStreamParser(string(peekbytes))
+
+							// Assume is done, check that the following 3 objects/operands are valid.
+							isDone := true
+							for i := 0; i < 3; i++ {
+								op, isOp, err := dummyParser.parseObject()
+								if err != nil && err != io.EOF {
+									continue
+								}
+								if err == io.EOF {
+									break
+								}
+								if isOp && !isValidOperand(op.String()) {
+									isDone = false
```

Comment:
Shouldn't there be a `break` here as well as it's already established that the peeked data does not contain valid operands?